### PR TITLE
chronicle - re-submitted

### PR DESCRIPTION
**Pull request by cc32d9**
re-submitting the Chronicle grant application

### CODE REVIEW

**Comment by tedcahalleos** [comment]
Your grant application has been approved by the grant committee and will now be processed by the assigned grant evaluators.  Please work with them for approval before proceeding to development of your milestones.

**Comment by arhag** [comment]
@cc32d9 

Can you resolve the conflicts in your current branch? That will create a new commit in your branch which allows the diff in this PR to introduce the whole document again, thus making it possible to comment on any of the lines.

Sorry for all the trouble with this re-submission.

**Comment by cc32d9** [comment]
done, now should merge. Thanks

File: applications/chronicle.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+- **Costs:** 8,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | Apache 2.0 |
+| 0b. | Documentation | Updated Chronicle README and the Tutorial to reflect all the recent changes |
+| 0c. | Testing Guide | Tests scripts and guides will be part of documentation |
+| 1. | Update dependencies | Update Chronicle to work with latest releases of dependency libraries (Boost.org and EOSIO/Mandel libraries).|  
+| 2. | Ubuntu packages | Crate package build scripts; publish the packages for public download |   
+| 3. | JavaScript NPM | Redesign the asynchronous mode to avoid memory overflow |
```
Comment (larryk85) [inline applications/chronicle.md:166]: What exactly does this mean?

File: applications/chronicle.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+### Milestone 2 — Research
+
+- **Estimated duration:** 2 month
+- **FTE:**  0.25
+- **Costs:** 8,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 1 | Alternative DB backend | Current Chronicle uses EOSIO Chainbase for its data. An alternative, more lightweight and less demanding, backend needs to be researched, and possibly taken for the next release. |
```
Comment (larryk85) [inline applications/chronicle.md:178]: What is the deliverable on this milestone? Is this just a proposed DB backend or a fully implemented alternative?

**Review by larryk85 — CHANGES_REQUESTED** [review]

File: applications/chronicle.md (hunk 1 above)
Comment (cc32d9) [reply applications/chronicle.md:166]: 1. update the code and git submodules to work with the current appbase/abieos/fc, and current Boost. Last time I trid, there was no simple upgrade path, so it needs work.
2. there's no packaging as of now, so we have to build from sources every time. Needs work to make proper deb package scripts
3. the NPM is overflowing the nodejs memory in asynchronous mode, as it cannot pause the source flow. So it needs a redesign.

Comment (larryk85) [reply applications/chronicle.md:166]: Gotcha. Also sorry about that I should've been more explicit. I just meant the NPM part.

**Review by larryk85 — APPROVED** [review]

**Review by 766C6164 — APPROVED** [review]
Approved. Milestones seems reasonable, competence and capacity of team to execute raises no questions. It would be fantastic to include Chronicle to integration test with Mandel for SHIP validation and to have a scaffold of architecture, that most of the app developers will find very helpful.

**Review by arhag — APPROVED** [review]

**Pull request merged**